compiler/src/iree/compiler/Codegen/SPIRV/KernelConfig.cpp

```diff
@@ -1632,6 +1632,23 @@ static LogicalResult setSPIRVOpConfig(const spirv::TargetEnv &targetEnv,
 static LogicalResult setConfigForKernel(const spirv::TargetEnv &targetEnv,
                                         IREE::HAL::ExecutableExportOp exportOp,
                                         func::FuncOp funcOp) {
+  // First check whether we already have workgroup count set--it's a "contract"
+  // to indicate that we should bypass all tiling and distribution to go down
+  // just the most basic lowering flow.
+  if (Block *body = exportOp.getWorkgroupCountBody()) {
+    auto retOp = cast<IREE::HAL::ReturnOp>(body->getTerminator());
+    // For scalar dispatch cases--using just one thread of one workgroup.
+    auto isOne = [](Value value) { return matchPattern(value, m_One()); };
```

Comment:
Do we need this check? as long as the workgroup count region is specified we should just do the default lowering IMO

Reply:
Just being conservative and not enable a lot of cases and see a surprise in the future--we can always drop later if other needs arise. :)